docs/Getting Started/Debian/Debian Bullseye Root on ZFS.rst

```diff
@@ -0,0 +1,1117 @@
+
+#. Create a volume dataset (zvol) for use as a swap device::
+
+     zfs create -V 4G -b $(getconf PAGESIZE) -o compression=zle \
+         -o logbias=throughput -o sync=always \
+         -o primarycache=metadata -o secondarycache=none \
+         -o com.sun:auto-snapshot=false rpool/swap
+
+   You can adjust the size (the ``4G`` part) to your needs.
+
+   The compression algorithm is set to ``zle`` because it is the cheapest
+   available algorithm. As this guide recommends ``ashift=12`` (4 kiB
+   blocks on disk), the common case of a 4 kiB page size means that no
+   compression algorithm can reduce I/O. The exception is all-zero pages,
+   which are dropped by ZFS; but some form of compression has to be enabled
+   to get this behavior.
+
+#. Configure the swap device:
+
+   **Caution**: Always use long ``/dev/zvol`` aliases in configuration
+   files. Never use a short ``/dev/zdX`` device name.
+
+   ::
+
+     mkswap -f /dev/zvol/rpool/swap
+     echo /dev/zvol/rpool/swap none swap discard 0 0 >> /etc/fstab
+     echo RESUME=none > /etc/initramfs-tools/conf.d/resume
+
+   The ``RESUME=none`` is necessary to disable resuming from hibernation.
+   This does not work, as the zvol is not present (because the pool has not
+   yet been imported) at the time the resume script runs. If it is not
+   disabled, the boot process hangs for 30 seconds waiting for the swap
+   zvol to appear.
+
+#. Enable the swap device::
+
+     swapon -av
+
+Step 8: Full Software Installation
+----------------------------------
+
+#. Upgrade the minimal system::
+
+     apt dist-upgrade --yes
+
+#. Install a regular set of software::
+
+     tasksel
+
+#. Optional: Disable log compression:
+
+   As ``/var/log`` is already compressed by ZFS, logrotate’s compression is
+   going to burn CPU and disk I/O for (in most cases) very little gain. Also,
+   if you are making snapshots of ``/var/log``, logrotate’s compression will
+   actually waste space, as the uncompressed data will live on in the
+   snapshot. You can edit the files in ``/etc/logrotate.d`` by hand to comment
+   out ``compress``, or use this loop (copy-and-paste highly recommended)::
+
+     for file in /etc/logrotate.d/* ; do
+         if grep -Eq "(^|[^#y])compress" "$file" ; then
+             sed -i -r "s/(^|[^#y])(compress)/\1#\2/" "$file"
+         fi
+     done
+
+#. Reboot::
+
+     reboot
+
+Step 9: Final Cleanup
+---------------------
+
+#. Wait for the system to boot normally. Login using the account you
+   created. Ensure the system (including networking) works normally.
+
+#. Optional: Delete the snapshots of the initial installation::
+
+     sudo zfs destroy bpool/BOOT/debian@install
+     sudo zfs destroy rpool/ROOT/debian@install
+
+#. Optional: Disable the root password::
+
+     sudo usermod -p '*' root
+
+#. Optional (but highly recommended): Disable root SSH logins:
+
+   If you installed SSH earlier, revert the temporary change::
+
+     vi /etc/ssh/sshd_config
```

Comment:
Let's remain sudo as in other parts of this section
```suggestion
     sudo vi /etc/ssh/sshd_config
```

Reply:
we don't do this anywhere else in the tutorial though, so that's a bit strange.

Reply:
It may be a copy-paste from Ubuntu how-tos. Another way is to remove `sudo` entirely from this section, I'm ok with both ways for now.

Reply:
i think i would favor that... but let's not make perfection the enemy of good here. :)